internal/infrastructure/kubernetes/proxy/resource_provider.go

```diff
@@ -72,6 +72,10 @@ func NewResourceRender(envoyNamespace, controllerNamespace, dnsDomain string, in
 }
 
 func (r *ResourceRender) Name() string {
+	if r.GatewayNamespaceMode {
+		return GetGatewayName(r.infra)
```

Comment:
imo, this is something that should be set in the gateway api layer which should directly set the `Name` field within `ProxyInfra`

Reply:
Right now, the infra name is actually the irKey.
Change that will be a huge mass, which should be do in a follow PR.

Reply:
only for this case though https://github.com/envoyproxy/gateway/blob/0815cdc695568e1b9fb66976df7daa87ed3bd1df/internal/gatewayapi/translator.go#L320

Reply:
hope https://github.com/envoyproxy/gateway/pull/6135/commits/426d8d98e47191fd723fd43e9438a9d9045512b6 is what you want.